perf/terraform/modules/long_lived/main.tf

```diff
@@ -0,0 +1,111 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.67.0"
+    }
+  }
+}
+
+variable "region" {
+  description = "The AWS region of the provider"
+}
+
+variable "ami" {
+  description = "The Amazon Machine Image to use"
+}
+
+locals {
+  availability_zone = "${var.region}a"
+}
+
+resource "aws_vpc" "perf" {
+  cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "perf" {
+  vpc_id                  = aws_vpc.perf.id
+  cidr_block              = "10.0.0.0/16"
+  availability_zone       = local.availability_zone
+  map_public_ip_on_launch = true
+}
+
+resource "aws_internet_gateway" "perf" {
+  vpc_id = aws_vpc.perf.id
+}
+
+resource "aws_route_table" "perf" {
+  vpc_id = aws_vpc.perf.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.perf.id
+  }
+}
+
+resource "aws_route_table_association" "perf" {
+  subnet_id      = aws_subnet.perf.id
+  route_table_id = aws_route_table.perf.id
+}
+
+resource "aws_security_group" "restricted_inbound" {
+  name        = "restricted_inbound"
+  description = "Allow SSH and port 4001 inbound traffic (TCP and UDP), allow all outbound traffic"
+  vpc_id      = aws_vpc.perf.id
+
+  # ICMP
+  ingress {
+    from_port   = -1
+    to_port     = -1
+    protocol    = "icmp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  # SSH (TCP)
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port   = 1
+    to_port     = 65535
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port   = 1
+    to_port     = 65535
+    protocol    = "udp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_launch_template" "perf" {
+  name          = "perf-node"
+  image_id      = var.ami
+  instance_type = "m5n.8xlarge"
```

Comment:
Currently this pull request uses [`m5n.8xlarge` machines with 25 Gibps](https://instances.vantage.sh/aws/ec2/m5n.8xlarge?selected=m5n.8xlarge&region=us-east-1&os=linux&cost_duration=hourly&reserved_term=Standard.noUpfront).

A couple of considerations:

- AWS offers either exactly XX or "up to" XX network performance. For what this guarantee is worth, I suggest, for the sake of reproducibility to choose the exactly XX types. (In case someone has more expertise here, please speak up.)
- No matter what instance type we choose, it will not be representative of the most libp2p deployments. At this point libp2p covers the whole space from a small laptop to huge bare metal machines.
- Thus far I have chosen an instance type such that the underlying hardware is not the bottleneck of the benchmark. In other words that the bottleneck shows up to be in the libp2p implementations itself.
- When it comes to network bandwidth this is a double-edged sword. I.e. high bandwidth requires large send and receive buffers. Large send and receive buffers make us vulnerable to DOS attacks.
- We could as well use the [general purpose `m5n.8xlarge` (note the missing `n`) instance type](https://instances.vantage.sh/aws/ec2/m5a.8xlarge). This would give us 10 Gibps instead of 25 Gibps at a slightly lower price.

With the above in mind, I opted to keep the instance type (`m5n.8xlarge`) as is.